data_prep/project_context/context_introspector.py

```diff
@@ -0,0 +1,202 @@
+"""Class to retrieve context from introspector for
+better prompt generation."""
+
+import os
+
+from data_prep import introspector
+from experiment import benchmark as benchmarklib
+
+
+class ContextRetriever:
+  """Class to retrieve context from introspector for
+  better prompt generation."""
+
+  def __init__(self, benchmark: benchmarklib.Benchmark):
+    """Constructor."""
+    self._benchmark = benchmark
+
+  def get_embeddable_declaration(self) -> str:
```

Comment:
i was thinking of making those public functions, in case some consumer of the api wanted to get the type-list / declaration separately. i cant think of a situation where they might want to be separated at the moment though. i can move these to private